patches/chromium/feat_add_storage_quota_capability_for_partitions.patch

```diff
@@ -0,0 +1,232 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: George Joseph <kottackal.george@gmail.com>
+Date: Tue, 27 Dec 2022 18:58:16 +0000
+Subject: feat: add storage quota capability for partitions
+
+This patch adds the capability for a storage quota
+to be set on partitions.This capability will be
+exercised only when physical storage based
+partitions are used.
+
+diff --git a/content/browser/storage_partition_impl.cc b/content/browser/storage_partition_impl.cc
+index bdc6a64ae010180102ebda39f1612604017fdd9f..40624cc28c594bab8aae0c01b0cc03a462152189 100644
+--- a/content/browser/storage_partition_impl.cc
++++ b/content/browser/storage_partition_impl.cc
+@@ -1221,6 +1221,10 @@ void StoragePartitionImpl::Initialize(
+       base::BindRepeating(&StoragePartitionImpl::GetQuotaSettings,
+                           weak_factory_.GetWeakPtr()));
+   quota_manager_ = quota_context_->quota_manager();
++
++  // Set a startup quota if present.
++  quota_manager_->SetStartupQuota(config_.quota());
```

Comment:
Is this even necessary? Can we avoid this patch altogether if we call SetQuotaSettings just after creating StoragePartition?

Reply:
> You should be able to use this here:
> 
> ```
> if (config_.quota())
>   quota_manager->SetQuotaSettings(storage::GetHardCodedSettings(config_.quota()));
> ```

Since the SetQuotaSettings call can be done only on the IOThread, I tried the snippet below - the call did succeed, however,  the static QuotaManagerImpl::GetVolumeInfo member was still getting called and hence the navigator.storage.estimate() call was returning the free space count on disk.


```
  base::WeakPtrFactory<storage::QuotaManager> quota_weak_factory_{quota_manager_.get()};
  GetIOThreadTaskRunner({})->PostTask(
          FROM_HERE, base::BindOnce(&storage::QuotaManager::SetQuotaSettings,
          quota_weak_factory_.GetWeakPtr(),storage::GetHardCodedSettings(config_.quota())));


```